docs/querying/caching.md

```diff
@@ -82,3 +82,19 @@ Note that the task executor processes only support caches that keep their data l
 This restriction exists because the cache stores results at the level of intermediate partial segments generated by the
 ingestion tasks. These intermediate partial segments will not necessarily be identical across task replicas, so
 remote cache types such as `memcached` will be ignored by task executor processes.
+
+## Unsupported queries
+
+Query caching is not available for following:
+- Queries, that involve a `union` datasource, do not support result-level caching. Refer to the 
+[related github issue](https://github.com/apache/druid/issues/8713) for details. Top level union SQL queries can still 
+be cached at result-level since they are not translated into a `union` datasource. Refer to [Union SQL](sql.md#UNION ALL) 
+for more details on how union SQL queries work.
+- Queries, that involve an `Inline` datasource or a `Lookup` datasource, do not support any caching. 
+- Queries, with a sub-query in them, do not support any caching though the output of sub-queries itself may be cached. 
+Refer to the [Query execution](query-execution.md#query) page for more details on how sub-queries are executed.
+- Join queries do not support any caching on the broker [More details](https://github.com/apache/druid/issues/10444).
+- GroupBy v2 queries do not support any caching on broker [More details](https://github.com/apache/druid/issues/3820).
```

Comment:
The linked issue only refers to segment-level caching.

I could be wrong but I believe GroupBy V2 does support results-level caching?

Reply:
It can support result level caching but right now all the caching is disabled for Group By V2 on broker. Result-level caching was introduced after disabling the broker caching for GroupBy V2. So it result-level caching remained disabled as well though there was no reason for it as such.